x/stakeibc/types/message_clear_balance.go

```diff
@@ -0,0 +1,53 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+const TypeMsgClearBalance = "clear_balance"
+
+var _ sdk.Msg = &MsgClearBalance{}
+
+func NewMsgClearBalance(creator string, chainId string, amount uint64, channelId string) *MsgClearBalance {
+	return &MsgClearBalance{
+		Creator:   creator,
+		ChainId:    chainId,
+		Amount: amount,
+		Channel: channelId,
+	}
+}
+
+func (msg *MsgClearBalance) Route() string {
+	return RouterKey
+}
+
+func (msg *MsgClearBalance) Type() string {
+	return TypeMsgClearBalance
+}
+
+func (msg *MsgClearBalance) GetSigners() []sdk.AccAddress {
+	creator, err := sdk.AccAddressFromBech32(msg.Creator)
+	if err != nil {
+		panic(err)
+	}
+	return []sdk.AccAddress{creator}
+}
+
+func (msg *MsgClearBalance) GetSignBytes() []byte {
+	bz := ModuleCdc.MustMarshalJSON(msg)
+	return sdk.MustSortJSON(bz)
+}
+
+func (msg *MsgClearBalance) ValidateBasic() error {
+	_, err := sdk.AccAddressFromBech32(msg.Creator)
+	if err != nil {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid creator address (%s)", err)
+	}
+	// TODO: add validation
```

Comment:
I know it doesn't really matter (not sure what the attack vector would be for this function), but feels _slightly_ safer to restrict this as an admin message, what do you think? 

The specific thing I'm paranoid about is if someone can time the message to cause an ICA timeout (or something else funny) that would cause the transaction to fail

The downside is it's fairly cumbersome to make function calls from the admin address (cold storage), and we might want to frequently call this function

Maybe we should make a "ValidateTrustedAddress" checker that has a slightly broader scope of people who can call it? And that has access to this function (maybe something like rebalance validators too?)

Reply:
Per offline discussion, risk/reward of exposing this is not in our favor. Going to gate it for now